src/utils.ts

```diff
@@ -538,34 +538,62 @@ export async function getDelegatesBySpace(
   const spaceIn = ['', space];
   if (space.includes('.eth')) spaceIn.push(space.replace('.eth', ''));
 
+  type Delegation = {
+    delegator: string;
+    delegate: string;
+    space: string;
+    timestamp: number;
+  };
   const PAGE_SIZE = 1000;
-  let result = [];
-  let page = 0;
+  let result: Delegation[] = [];
   const params: any = {
     delegations: {
       __args: {
         where: {
           space_in: spaceIn
         },
         first: PAGE_SIZE,
-        skip: 0
+        skip: 0,
+        orderBy: 'timestamp',
+        orderDirection: 'asc'
       },
       delegator: true,
       space: true,
-      delegate: true
+      delegate: true,
+      timestamp: true
     }
   };
   if (snapshot !== 'latest') {
     params.delegations.__args.block = { number: snapshot };
   }
 
+  const isSameDelegate = (a: Delegation, b: Delegation) => {
+    return (
+      a.delegator === b.delegator &&
+      a.delegate === b.delegate &&
+      a.space === b.space
+    );
+  };
+
   while (true) {
-    params.delegations.__args.skip = page * PAGE_SIZE;
+    const pivot = result[result.length - 1];
+    params.delegations.__args.where.timestamp_gte = pivot?.timestamp || 0;
+
+    let pageDelegations: Delegation[] =
+      (await subgraphRequest(subgraphUrl, params)).delegations || [];
+
+    // Removing duplicates results between pages because of the _gte filter
+    if (pivot) {
+      // List of delegation that can potentially be duplicates in the new page results
+      const pivotDelegations = result.filter((d) => isSameDelegate(d, pivot));
```

Comment:
There can be multiple records with the same `timestamp` better to do filter duplicates once pagination is completed, 

I am also worried about what happens if there are 1000 records with the same timestamp. (highly unlikely)

Reply:
Yes, this implementation loosely assumes that there will not be more than 1000 duplicates

Records are sorted by timestamp, so checking for duplicate on each page will be more easier than doing it at once at the end, on an array of 5000 items.